### PR TITLE
feat: :alembic: add check-spelling.yml gh action

### PR DESCRIPTION
**Pull request by peelar**
**Featuring**:
- cspell GitHub action
- `check-spelling` global command that runs on all files
- `cspell` command in lint-staged that checks if there are typos before committing
- dictionary of commonly used proper names

### CODE REVIEW

File: cspell.json [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    "typeorm",
+    "Undiscounted",
+    "Upstash",
+    "urql",
+    "usehooks",
+    "uvicorn",
+    "VERCEL",
+    "Windi",
+    "windicss",
+    "xstate",
```
Comment (lkostrowski) [inline cspell.json:89]: are these actual wordings? I dont see where we have xstate

**Review by lkostrowski — CHANGES_REQUESTED** [review]

**Comment by lkostrowski** [comment]
@peelar please add to Appstore repo too

File: cspell.json (hunk 1 above)
Comment (peelar) [reply cspell.json]: I checked it to only contain Saleor-specific names and integrations, as it will not scan package.json anymore

**Comment by peelar** [comment]
> @peelar please add to Appstore repo too

@lkostrowski done

**Review by lkostrowski — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -4,6 +4,7 @@
   "scripts": {
     "build": "turbo run build",
     "check-deps": "syncpack list-mismatches",
+    "check-spelling": "cspell **/*.{jsx,tsx,js,ts}",
```
Comment (krzysztofwolski) [inline package.json:7]: Missing md files

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]

**Review by krzysztofwolski — APPROVED** [review]

**Pull request merged**